src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlTest.java

```diff
@@ -175,7 +175,7 @@ public void testRegExpParentInSubpackageIsConsidered() {
         assertEquals("Invalid package", icCommonRegexpParen, icRootRegexpParent
                 .locateFinest("com.kazgroup.courtlink.common.api"));
         assertEquals("Invalid package", icCommonRegexpParen, icRootRegexpParent
-                .locateFinest("com.kazgroup.courtlink.comon.api"));
+                .locateFinest("com.kazgroup.courtlink.common.api"));
     }
```

Comment:
The second asserts becomes meaningless. I think it is possible to replace "сomon" with "other".

Reply:
The mis-spelling here is on purpose. Look at the regular expression at https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlTest.java#L40 .
It is `m+` meaning it can be `comon`, `common`, commmon`, etc...

Reply:
The name of the package can be anything?
Can it be
`com.kazgroup.courtlink.bot.api` and `com.kazgroup.courtlink.boot.api`
And the regexp should be 
```
private final ImportControl icCommonRegexpParen = new ImportControl(icRootRegexpParent,
"bo+t", true);
```
In this case both the spelling and the test will be OK.

Reply:
I can make that change if requested (it's hard to tell which items are directions and which are just conversations).

For the time being, I've dropped this (because it was the simplest path forward).